### PR TITLE
Split permission requests for Camera & Gallery

### PR DESCRIPTION
**Pull request by lizraeli**

### CODE REVIEW

**Comment by sryze** [comment]
Hi @lizraeli, why did you close this PR? Was there a reason? I too think that this package needs to request specific permissions based on whether requesting to open the photo library or camera. Personally I only use it to ask users to pick a photo from their library, not camera, so I think this patch is useful.

**Comment by Tatandiaz13** [comment]
I did have this error #563 and this changes  work for me !

**Comment by jacquesdev** [comment]
@lizraeli - Can we please get this issue re-opened and merged? The issue seems like a bug, to be honest, and this will fix it.

**Comment by lizraeli** [comment]
@jacquesdev I've reopened the PR. I am happy to merge this once it gets reviewed.

File: android/src/main/java/com/imagepicker/ImagePickerModule.java [hunk 1]
```diff
@@ -547,10 +553,16 @@ private boolean permissionsCheck(@NonNull final Activity activity,
...
 
-    final boolean permissionsGrated = writePermission == PackageManager.PERMISSION_GRANTED &&
-            cameraPermission == PackageManager.PERMISSION_GRANTED;
+    boolean permissionsGranted = false;
 
-    if (!permissionsGrated)
+    switch (requestCode) {
+      case REQUEST_PERMISSIONS_FOR_LIBRARY:
+        permissionsGranted = writePermission == PackageManager.PERMISSION_GRANTED;
+      case REQUEST_PERMISSIONS_FOR_CAMERA:
```
Comment (Johan-dutoit) [inline android/src/main/java/com/imagepicker/ImagePickerModule.java]: You're missing a break statement here, which causes the wrong 'permission' to be checked when selecting image from library

**Comment by chengsam** [comment]
I wonder why a 5k starred library would have this bug (or feature)?

File: android/src/main/java/com/imagepicker/ImagePickerModule.java (hunk 1 above)
Comment (lizraeli) [reply android/src/main/java/com/imagepicker/ImagePickerModule.java]: @Johan-dutoit fixed!

**Comment by genuinezack** [comment]
Agreed this would be a good feature to have cc @Johan-dutoit and @janicduplessis

**Comment by kayzenkayzen** [comment]
Please merge!

**Comment by KianooshSoleimani** [comment]
any solution? split permissions is need

**Pull request merged**